lib/practice_exercises.rb

```diff
@@ -1,13 +1,60 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
```

Comment:
👍  However you are using `.downcase` and `.gsub` each of which creates a new string, so O(n) space complexity.  Your `reverse_string` method has similar issues.